src/c4/C4_MPI.i.hh

```diff
@@ -0,0 +1,98 @@
+//----------------------------------*-C++-*----------------------------------//
+/*!
+ * \file   c4/C4_MPI.i.hh
+ * \author Alex R Long
+ * \date   Mon Aug 21 07:47:01 2017
+ * \brief  C4 MPI standard implementations.
+ * \note   Copyright (C) 2017 Los Alamos National Security, LLC.
+ *         All rights reserved. */
+//---------------------------------------------------------------------------//
+
+#ifndef c4_C4_MPI_i_hh
+#define c4_C4_MPI_i_hh
+
+#ifdef C4_MPI
+
+namespace rtt_c4 {
+
+//---------------------------------------------------------------------------//
+template <typename T>
+DLL_PUBLIC_c4 void send_is_custom(C4_Req &request, const T *buffer, int size,
+                                  int destination, int tag) {
+
+  Require(!request.inuse());
+
```

Comment:
What happens if `size=0` or `buffer=nullptr`?  Can DbC be added for those cases? Is special logic needed?  (here and new functions below).